src/main.cpp

```diff
@@ -4512,6 +4518,15 @@ void static ProcessGetData(CNode* pfrom, const Consensus::Params& consensusParam
                         // else
                             // no response
                     }
+                    else if (inv.type == MSG_CMPCT_BLOCK)
+                    {
+                        // Note that while we will respond to all MSG_CMPCT_BLOCK requests here,
```

Comment:
There is no DoS score involved in a request for which we did not send an inv (ie something which is buried in our chain), only for those which request something with !BLOCK_HAVE_DATA. For those, we definitely wont even send a CMPCTBLOCK, because its covered by the surrounding if statement. 

As for not sending the compact block if the block is >= 10 deep, yes, I can add an if statement and simply not respond to the getdata for those.